src/main/scala/org/broadinstitute/dsde/workbench/sam/service/PolicyEvaluatorService.scala

```diff
@@ -71,21 +75,39 @@ class PolicyEvaluatorService(
       res <- resourceTypes.get(resource.resourceTypeName).traverse {
         rt =>
 
-          // we can only use this approach if the resource type isn't auth-domain constrainable
-          if (!rt.isAuthDomainConstrainable) {
-
-            // compute the list of roles that contain the action requested
-            val roleNamesWithAction = rt.roles.filter(_.actions.contains(action)).map(_.roleName)
+          val roleNamesWithAction = rt.roles.filter(_.actions.contains(action)).map(_.roleName)
+          val hasAction = directMemberHasActionOnResource(resource, roleNamesWithAction, action, userId)
 
+          // if the resource type OR action are NOT auth domain constrainable... just return
+          if (!rt.isAuthDomainConstrainable || !rt.actionPatterns.exists( ap => ap.authDomainConstrainable && ap.matches(action)) ) {
+            hasAction
+          } else {
             for {
-              // get the policies for this resource and filter to ones that contain the roles with the action, or the action directly
-              policies <- accessPolicyDAO.listAccessPolicies(resource).map(_.toList.filter(p => p.roles.intersect(roleNamesWithAction).nonEmpty || p.actions.contains(action)))
-              members = policies.flatMap(_.members)
-            } yield members.contains(userId)
-          } else IO.pure(false)
+              // first, check if the resource in question HAS an auth domain (likely not) and if so continue [TERRA]
+              authDomainsResult <- accessPolicyDAO.loadResourceAuthDomain(resource)
+              res2 <- authDomainsResult match {
+                case LoadResourceAuthDomainResult.NotConstrained | LoadResourceAuthDomainResult.ResourceNotFound =>
+                  hasAction
+                case LoadResourceAuthDomainResult.Constrained(authDomains) =>
+                  // if there is more than one group, just fallback
+                  if (authDomains.size > 1) {
+                    IO.pure(false)
```

Comment:
since isGroupMember is better, I don't think you need to fallback in this case:
```authDomains.traverse(directoryDAO.isGroupMember(_, userId)).map(_.forall(identity))```

Reply:
after talking...  going to save this for a future PR, we're at the point where we could make the Shallow check... full, but still fast.  But also by that point all the LDAP stuff will be not long for this world